### PR TITLE
Add directive

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* [X] Has `CHANGELOG.md` been updated?
* [X] Are the files formatted correctly?
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Ability to delay native start using a compile directive.

## Related PRs or issues

[AB#69564](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/69564) #326

### CODE REVIEW

**Review by Jamminroot — DISMISSED** [review]

**Review by russelarms — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### App Center Push
 
 * **[Bug fix]** Fix missing `NotifyPushNotificationReceived` callback if push notification is received from the background.
+* **[Enhancement]** Add ability to delay native start using a scripting define symbol `DONT_USE_NATIVE_STARTER`.
```
Comment (guperrot) [inline CHANGELOG.md]: Should this be prefixed by `APPCENTER_` ? It's not a standard one.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### App Center Push
 
 * **[Bug fix]** Fix missing `NotifyPushNotificationReceived` callback if push notification is received from the background.
+* **[Enhancement]** Add ability to delay native start using a scripting define symbol `APPCENTER_DONT_USE_NATIVE_STARTER`.
```
Comment (jaeklim) [inline CHANGELOG.md]:
```suggestion
* **[Improvement]** Add ability to delay native start using a scripting define symbol `APPCENTER_DONT_USE_NATIVE_STARTER`.
```

File: Assets/AppCenter/Editor/AppCenterPreBuild.cs [hunk 3]
```diff
@@ -36,12 +36,16 @@ public void OnPreprocessBuild(BuildTarget target, string path)
             {
                 FirebaseDependency.SetupPush();
             }
+#if !APPCENTER_DONT_USE_NATIVE_STARTER
```
Comment (jaeklim) [inline Assets/AppCenter/Editor/AppCenterPreBuild.cs:39]: Consider change the macro to `APPCENTER_USE_NATIVE_STARTER`

Comment (annakocheshkova) [reply Assets/AppCenter/Editor/AppCenterPreBuild.cs:39]: @jaelim-ms disagree on this one. This is an optional directive for a very small fraction of our customers who want to delay the start. So if we change to `APPCENTER_USE_NATIVE_STARTER`, almost everyone will have to specify it.

**Review by guperrot — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**